doc/build/build_from_source.md

```diff
@@ -139,3 +143,129 @@ And if you set WITH_SWIG_PY=ON, you have to install related python predict api a
 ```bash
 pip install <path to install>/opt/paddle/share/wheels/*.whl
 ```
+## <span id="mac">Building on Mac OS X</span>
+
+### Prerequisites
+This guide is based on Mac OS X 10.11 (El Capitan). Note that if you are running an up to date version of OS X, 
+you will already have Python 2.7.10 and Numpy 1.8 installed.
+
+The best option is to use the package manager homebrew to handle installations and upgrades for you.
+To install homebrew, first open a terminal window (you can find Terminal in the Utilities folder in Applications), and issue the command:
+
+```bash
+# install brew
+/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+# install pip
+easy_install pip
+```
+
+### Install Dependencies
+
+- **CPU Dependencies**
+
+```bash
+# Install fundamental dependents 
+brew install glog gflags cmake protobuf openblas
+
+# Install google test on Mac OS X
+# Download gtest 1.7.0
+wget https://github.com/google/googletest/archive/release-1.7.0.tar.gz
+tar -xvf googletest-release-1.7.0.tar.gz && cd googletest-release-1.7.0
+# Build gtest
+mkdir build && cmake ..
+make
+# Install gtest library
```

Comment:
When PaddlePaddle linked with Openblas on MAC OS X, run unit test triggers "35: BLAS : Program is Terminated. Because you tried to allocate too many memory regions. ", I guess the problem is from openblas.